test/helpers/table.js

```
@@ -17,7 +18,41 @@ export async function getTableItem (dynamo, tableName, key) {
 }
 
 /**
- * @param {import("@aws-sdk/client-dynamodb").DynamoDBClient} dynamo
+ * @param {import('@aws-sdk/client-dynamodb').DynamoDBClient} dynamo
+ * @param {string} tableName
+ * @param {Record<string, import('@aws-sdk/client-dynamodb').Condition>} keyConditions
+ * @param {object} [options]
+ * @param {string} [options.indexName]
+ */
+export async function pollQueryTable (dynamo, tableName, keyConditions, options = {}) {
+  const cmd = new QueryCommand({
+    TableName: tableName,
+    KeyConditions: keyConditions,
+    IndexName: options.indexName,
+  })
+
+  let response
+  try {
+    response = await pRetry(async () => {
+      const r = await dynamo.send(cmd)
+      if (r.$metadata.httpStatusCode === 404 || !r.Count) {
+        throw new Error('not found in dynamoDB yet')
+      }
+      return r
+    }, {
+      maxTimeout: 2000,
+      minTimeout: 1000,
+      retries: 100
+    })
+  } catch {}
```

I'd let this throw - if we don't find after 100 retries it ain't coming.